File: AngularERP/src/app/services/accounting/AccountChartService.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { environment } from "../../../environments/environment";
import {AccountChart} from '../../models/accounting/basic-entry/bank/AccountChart'

@Injectable({
  providedIn: 'root'
})
export class AccountChartService {

  constructor(private http:HttpClient) { }

  saveAccChart(acctchart:AccountChart){
    console.log(acctchart);
    return this.http.post(environment.apiUrl+'/accountchart/saveAccountChart',acctchart);
 }
 getAllBranchName(){

  return this.http.get(environment.apiUrl+'/accountchart/getAllBranch');

}
getAllCurrencies(){
  return this.http.get(environment.apiUrl+'/accountchart/getAllCurrencyList');
}

//.................
getAllGroupAccount(){
  //return this.http.get(environment.apiUrl+'/basicEntry/getAllGroupAccount/compID/'+compId+'/lwGID/'+lowerGroupID);
  return this.http.get(environment.apiUrl+'/accountchart/getAllGroupAccount');
}
getAllAccountChart(){
//  return this.http.get(environment.apiUrl+'/basicEntry/getAllAccountChart/compId/'+compId +'/recivablepayble/'+recivablepayble+'/branchID/'+branchID);
  return this.http.get(environment.apiUrl+'/accountchart/getAllAccountChart');
 }
 updateAccChart(accountchart: AccountChart){
   return this.http.put(environment.apiUrl+'/accountchart/updateAccountChart', accountchart);
 }
 //compId:number,recivablepayble:number,branchID:number
///Transaction Service


}

File: ERP/src/main/java/com/boot/accounting/service/AccountChartService.java
package com.boot.accounting.service;

import java.util.List;

import com.boot.accounting.model.AccountChart;
import com.boot.accounting.model.AccountGroupType;
import com.boot.accounting.model.Branch;
import com.boot.accounting.model.CurrencyInfo;

public interface AccountChartService {
	AccountChart saveAccountChart(AccountChart accountChart);
	AccountChart updateAccountChart(AccountChart accountChart);
	boolean deleteByIdAccountChart(Long id);
	List<AccountChart> getAllAccountChart();
	List<Branch> getAllBranch();
	List<AccountGroupType> getAllAccountGroupType();
	List<CurrencyInfo> getAllCurrency();
}
